pkg/ctl/cmdutils/configfile.go

```diff
@@ -761,3 +763,84 @@ func NewDeleteIAMServiceAccountLoader(cmd *Cmd, sa *api.ClusterIAMServiceAccount
 
 	return l
 }
+
+// NewUpdateNodegroupLoader will load config or use flags for 'eksctl update nodegroup'.
+func NewUpdateNodegroupLoader(cmd *Cmd) ClusterConfigLoader {
+	l := newCommonClusterConfigLoader(cmd)
+
+	l.validateWithConfigFile = func() error {
+		var clusterConfig *api.ClusterConfig
+		var err error
+		if clusterConfig, err = eks.LoadConfigFromFile(l.ClusterConfigFile); err != nil {
+			return err
+		}
+
+		l := len(clusterConfig.ManagedNodeGroups)
+		if l < 1 {
+			return ErrMustBeSet("managedNodeGroups field")
+		}
+
+		var unsupportedFields []string
+		ng := clusterConfig.ManagedNodeGroups[0]
+		if unsupportedFields, err = validateSupportedConfigFields(*ng.NodeGroupBase, []string{"Name"}, unsupportedFields); err != nil {
+			return err
+		}
+
+		if unsupportedFields, err = validateSupportedConfigFields(*ng, []string{"NodeGroupBase"}, unsupportedFields); err != nil {
+			return err
+		}
+
+		logger.Warning("unchanged fields: the following fields remain unchanged; they are not supported by `eksctl update nodegroup`: %s", strings.Join(unsupportedFields[:], ", "))
```

Comment:
will this log even if that `unsupportedFields` array is empty?

Reply:
good shout, changing it to check `len` of `unsupportedFields` first!